commit abdeb4b5200e0afb05e6a7863c52d2fbe7029b47
Author: Florian Weimer <fweimer@redhat.com>
Date:   Tue May 20 19:36:02 2025 +0200

    support: Use const char * argument in support_capture_subprogram_self_sgid
    
    The function does not modify the passed-in string, so make this clear
    via the prototype.
    
    Reviewed-by: Carlos O'Donell <carlos@redhat.com>
    (cherry picked from commit f0c09fe61678df6f7f18fe1ebff074e62fa5ca7a)

diff --git a/support/capture_subprocess.h b/support/capture_subprocess.h
index 1ecbdfe4fc4aa123..f2765278d920839d 100644
--- a/support/capture_subprocess.h
+++ b/support/capture_subprocess.h
@@ -44,8 +44,7 @@ struct support_capture_subprocess support_capture_subprogram
 /* Copy the running program into a setgid binary and run it with CHILD_ID
    argument.  If execution is successful, return the exit status of the child
    program, otherwise return a non-zero failure exit code.  */
-int support_capture_subprogram_self_sgid
-  (char *child_id);
+int support_capture_subprogram_self_sgid (const char *child_id);
 
 /* Deallocate the subprocess data captured by
    support_capture_subprocess.  */
diff --git a/support/support_capture_subprocess.c b/support/support_capture_subprocess.c
index ffced8a89fca37a5..eb72a2c21cf99ee2 100644
--- a/support/support_capture_subprocess.c
+++ b/support/support_capture_subprocess.c
@@ -109,7 +109,7 @@ support_capture_subprogram (const char *file, char *const argv[])
    safely make it SGID with the TARGET group ID.  Then runs the
    executable.  */
 static int
-copy_and_spawn_sgid (char *child_id, gid_t gid)
+copy_and_spawn_sgid (const char *child_id, gid_t gid)
 {
   char *dirname = xasprintf ("%s/tst-tunables-setuid.%jd",
 			     test_dir, (intmax_t) getpid ());
@@ -181,7 +181,7 @@ copy_and_spawn_sgid (char *child_id, gid_t gid)
   ret = 0;
   infd = outfd = -1;
 
-  char * const args[] = {execname, child_id, NULL};
+  char * const args[] = {execname, (char *) child_id, NULL};
 
   status = support_subprogram_wait (args[0], args);
 
@@ -210,7 +210,7 @@ err:
 }
 
 int
-support_capture_subprogram_self_sgid (char *child_id)
+support_capture_subprogram_self_sgid (const char *child_id)
 {
   gid_t target = 0;
   const int count = 64;
